third_party/nvfuser/csrc/kernel_cache.cpp

```diff
@@ -222,6 +222,78 @@ std::vector<at::Tensor> FusionExecutorCache::runFusionWithInputs(
   return outputs;
 }
 
+std::string FusionExecutorCache::getCode(
+    FusionKernelRuntime* kernel_runtime,
+    bool intrinsic_code) const {
+  std::string kernel_code;
+  TORCH_CHECK(kernel_runtime != nullptr, "Invalid fusion definition!");
+  TORCH_CHECK(kernel_runtime->isCompiled(), "Fusion is not compiled!");
+
+  bool first_kernel = true;
+  for (const auto& exec : kernel_runtime->executors()) {
+    if (first_kernel) {
+      first_kernel = false;
+    } else {
+      kernel_code += "\n";
+    }
+    kernel_code += exec.kernelString();
+  }
+
+  if (intrinsic_code) {
+    const FusionExecutor& fe = kernel_runtime->executors()[0];
+    std::string full_code =
+        fe.getStructuredCode(kernel_code, fe.kernel()->indexType());
```

Comment:
Do all executors have the same index type? Even though I recently did some cleanup of the kernel index type, I'm still unsure here.

Reply:
I assumed the code I am calling the "intrinsic_code" which we print as the "full_kernel" is identical for each kernel in segmentation.  Is that an incorrect assumption?  I could also print the "full_kernel" for each kernel of segmentation but I thought that might be too verbose.

Reply:
`getStructuredCode` inserts a macro definition of the index type, which is either `int` or `int64_t`. The index type of a FusionExecutor is determined by the KernelArgumentHolder passed to `compileFusion`. I think it only holds the inputs to each segment of the original fusion, so it seems possible to have different segments to have different index types.

Reply:
For example, if only one segment uses a very large tensor that exceeds the 32-bit indexing limit, only that segment would have the index type defined as `int64_t`, and the rest of the segments should use `int`. I haven't verified this behavior, though.

Reply:
I'm just saying the printed code done in this PR may not be correct. Maybe it doesn't matter? Not sure what is expected.